quickstart/gcp-cloudshell-quickstart.py

```diff
@@ -88,20 +96,110 @@
+def ensure_requirements():
     if not PROJECT_ID:
         print('The PROJECT property has not been set.')
         print('Please run "gcloud config set project [PROJECT_ID]" to set the project.')
         print('See: https://cloud.google.com/sdk/gcloud/reference/config/set')
         print('')
         sys.exit(1)
 
+    ensure_google_libraries()
+    ensure_terraform()
+
+
+def ensure_google_libraries():
+    """A function that ensures Google libraries are installed. 
+
+    The function first tries to import the required Google libraries and if the required packages
+    are not installed, it will prompt the user to install these packages in the user's home directory. 
+    """
+
+    global INSTALL_GOOGLE_LIBRARIES
+
+    google_libraries_import = """
+import googleapiclient.discovery
+from google.cloud import kms_v1
+from google.cloud.kms_v1 import enums
+from google.api_core import exceptions as google_exc
+"""
+
+    try:
+        exec(google_libraries_import, globals())
+        print('Successfully imported Google libraries.\n')
+
+    except ImportError as err:
+        INSTALL_GOOGLE_LIBRARIES = True
+
+    if INSTALL_GOOGLE_LIBRARIES:
+        google_libraries = 'google-api-python-client grpc-google-iam-v1 google-cloud google-cloud-kms'
+
+        install_permission = input(
+            'One or more of the following Python packages are missing:\n'
+            f'  {google_libraries}\n\n'
+            'The script can install these packages in the user\'s home directory using the following command:\n' 
+            f'  (i.e. {sys.executable} -m pip install {google_libraries} --user)\n'
+            'Proceed? (y/n)? ').strip().lower()
+
+        if install_permission not in ('y', 'yes'):
+            print('Google libraries are required for deployment. Exiting...')
+            sys.exit(1)
+
+        install_cmd = f'{sys.executable} -m pip install {google_libraries} --user'
+        subprocess.check_call(install_cmd.split(' '))
+
+        # Recommended to clear cache after installing python packages for dynamic imports
+        importlib.invalidate_caches()
+
+        exec(google_libraries_import, globals())
+
+
+def ensure_terraform():
+    """A function that ensures Terraform version >= 0.12 is installed. 
+
+    The function first checks if Terraform is installed in the user's system and if Terraform
+    is not installed, or it is not version >= 0.12, it will prompt the user to install Terraform
+    in the user's home directory. 
+    """
+
+    global INSTALL_TERRAFORM
+    global TERRAFORM_BIN_PATH
+
+    path = shutil.which('terraform')
+
+    if path:
+        cmd = 'terraform -v'
+        # Run the command 'terraform -v' and use the first line as the Terraform version
+        terraform_version = subprocess.run(cmd.split(' '),  stdout=subprocess.PIPE).stdout.decode('utf-8').splitlines()[0]
+        
+        print(f'Found {terraform_version} in {path}.\n')
+
+        if 'v0.12.' not in terraform_version:
```

Comment:
You can check if v0.12. is in terraform_version, if so then just return the path.

Then you don't need the else and just proceed to install terrform without an extra if on L191.

Also returning the path will mean you avoid referencing the global variable.